src/mscorlib/src/System/Collections/Generic/List.cs

```diff
@@ -291,12 +290,14 @@ public int BinarySearch(T item, IComparer<T> comparer)
 
     
         // Clears the contents of List.
-        public void Clear() {
-            if (_size > 0)
+        public void Clear()
+        {
+            if (JitHelpers.ContainsReferences<T>() && _size > 0)
```

Comment:
There are other places in this file that call Array.Clear. We should fix them all.